packages/core/src/components/StepList/StepList.scss

```diff
@@ -0,0 +1,413 @@
+}
+
+.ds-c-step {
+  color: $color-muted;
+  counter-increment: step-counter;
+  margin-bottom: $step-margin;
+  padding-bottom: $step-margin;
+  position: relative;
+
+  &::before {
+    border: $step-list-number-border-size solid $color-muted;
+    border-radius: 50%;
+    box-sizing: border-box;
+    color: $color-muted;
+    content: counter(step-counter);
+    display: block;
+    font-size: $step-list-number-font-size-mobile;
+    height: $step-list-number-size-mobile;
+    left: -($step-list-number-margin + $step-list-number-size-mobile);
+    line-height: $step-list-number-size-mobile - $step-list-number-border-size * 2;
+    position: absolute;
+    text-align: center;
+    top: ($step-button-height - $step-list-number-size-mobile) / 2;
+    width: $step-list-number-size-mobile;
+
+    @media (min-width: $step-list-breakpoint) {
+      font-size: $step-list-number-font-size;
+      height: $step-list-number-size;
+      left: -($step-list-number-margin + $step-list-number-size);
+      line-height: $step-list-number-size - $step-list-number-border-size * 2;
+      padding-left: 1px;
+      top: ($step-button-height - $step-list-number-size) / 2;
+      width: $step-list-number-size;
+    }
+  }
+
+  &::after {
+    border-bottom: $step-border-width solid $border-color;
+    bottom: 0;
+    content: '';
+    display: block;
+    height: 1px;
+    left: -$step-list-padding-left;
+    position: absolute;
+    right: -$step-list-padding-h;
+  }
+
+  @media (min-width: $step-list-breakpoint) {
+    display: flex;
+    justify-content: space-between;
+  }
+}
+
+.ds-c-step--current,
+.ds-c-step--completed {
+  padding-bottom: $step-margin;
+
+  &::before {
+    border: 0 none;
+    color: $color-white;
+    line-height: $step-list-number-size-mobile;
+
+    @media (min-width: $step-list-breakpoint) {
+      line-height: $step-list-number-size;
+    }
+  }
+}
+
+.ds-c-step--current {
+  &::before {
+    background: $current-step-color;
+  }
+}
+
+.ds-c-step--completed {
+  .ds-c-step__substeps {
+    color: $color-base;
+  }
+
+  &::before {
+    background: $current-step-color;
+  }
+}
+
+.ds-c-step__content {
+  @media (min-width: $step-list-breakpoint) {
+    padding-right: $spacer-2;
+  }
+}
+
+.ds-c-step__title {
+  font-size: $h5-font-size;
+  line-height: $step-button-height;
+  margin: 0;
+
+  @media (min-width: $step-list-breakpoint) {
+    font-size: $h3-font-size;
+  }
+}
+
+.ds-c-step__actions {
+  font-size: $small-font-size;
+
+  .ds-c-step__content--with-content + & {
+    margin-top: $spacer-2;
+
+    @media (min-width: $step-list-breakpoint) {
+      margin-top: 0;
+    }
+  }
+
+  .ds-c-step--completed & a {
+    font-weight: $font-normal;
+  }
+
+  @media (min-width: $step-list-breakpoint) {
+    font-size: $base-font-size;
+    text-align: right;
+  }
+}
+
+.ds-c-step__completed-text {
+  display: inline-block;
+  line-height: $step-button-height - 12px;
+  margin-right: $spacer-2;
+  padding-top: 6px;
```

Comment:
Last nit, I promise: When the actions are collapsed, this padding is causing too much space to be above. We'd only want this padding when the actions aren't collapsed — since that centers the completed text with the title, which is nice.

Reply:
For the record, visual details aren't a nit to me. I like things pixel-perfect.